lib/charms/mongodb/v0/set_status.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+"""Code for handing statuses in the app and unit."""
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from ops.charm import CharmBase
+from ops.framework import Object
+from ops.model import StatusBase
+
+from config import Config
+
+# The unique Charmhub library identifier, never change it
+LIBID = "9b0b9fac53244229aed5ffc5e62141eb"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+class MongoDBStatusHanlder(Object):
```

Comment:
Typo

Reply:
If this class is only used to check if shards are active OR upgrading, why then not having two flags in the relation instead?

Reply:
Maybe I missed something; but can units' relation data be accessed by external applications?